Sources/NIOExtras/PCAPRingBuffer.swift

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+// MARK: NIOPCAPRingBuffer
+/// Storage for the most recent set of packets captured subject to constraints.
+/// Use `addFragment` as the sink to a `NIOWritePCAPHandler` and call `emitPCAP`
+/// when you wish to get the recorded data.
+public class NIOPCAPRingBuffer {
+    private var pcapFragments: CircularBuffer<ByteBuffer>
+    private var pcapCurrentBytes: size_t
+    private let maximumFragments: UInt
+    private let maximumBytes: size_t
+
+    /// Initialise the buffer, setting constraints.
+    /// Parameters:
```

Comment:
nit: you need `/// - Parameters:` for Xcode to render the documentation it correctly (here and elsewhere) 

fyi: you can use "⌘⌥/" on the function signature to generate the boilerplate for you